packages/core/src/network/network-peers.ts

```diff
@@ -102,22 +103,26 @@ class NetworkPeers {
       }
     }
 
-    this.peers[entryIndex] = newEntry
+    this.peers.set(id, newEntry)
   }
 
   // Get a random sample peers.
   public randomSubset(size: number, filter?: (peer: PeerId) => boolean): PeerId[] {
+    const peers = Array.from(this.peers.values())
```

Comment:
Beware that this changes semantics: if `size` argument equals to the number of peers, the `randomSubset` performs a random permutation in-place on the given array. The original behavior might be even considered a bug.

Reply:
Makes sense, indeed it was a bug.